providers/ethereum-provider/src/EthereumProvider.ts

```diff
@@ -316,15 +320,15 @@ export class EthereumProvider implements IEthereumProvider {
       } else if (event.name === "chainChanged") {
         this.setChainId(this.formatChainId(event.data));
       } else {
-        this.events.emit(event.name, event.data);
+        this.events.emit(event.name as any, event.data);
```

Comment:
Wouldn’t this always be `string`? Why the cast to any here?

Reply:
the `events.emit` is now typed to these events 
```
type Event =
    | "connect"
    | "disconnect"
    | "message"
    | "chainChanged"
    | "accountsChanged"
    | "session_delete"
    | "session_event"
    | "session_update"
    | "display_uri";
```
but technically you could also emit custom event